dapps/v2Explorer/src/utils/UniversalProvider.tsx

```diff
@@ -46,11 +53,29 @@ export async function createUniversalProvider() {
     });
 
     // Subscribe to session delete
-    universalProvider.on('session_delete', ({id, topic}) => {
-      console.log('session_delete', id, topic);
-    });
+    universalProvider.on(
+      'session_delete',
+      ({id, topic}: {id: string; topic: string}) => {
+        onSessionDisconnect?.({id, topic});
+        console.log('session_delete', id, topic);
+      },
+    );
+  } catch {
+    console.log('Error for connecting');
+  }
+}
+
+export function clearSession() {
+  universalProviderSession = undefined;
+  web3Provider = undefined;
+}
 
-    universalProviderSession = await universalProvider.connect({
+export async function createUniversalProviderSession(callbacks?: {
+  onSuccess?: () => void;
+  onFailure?: (error: any) => void;
+}) {
+  await universalProvider
```

Comment:
await and `.then` used, are both necessary? Where possible let's stick to `async/await` with `try/catch`

Reply:
yep, you are right, forgot to remove that await!